src/Framework/TaskHostParameters.cs

```diff
@@ -0,0 +1,113 @@
+        /// <value>The runtime identifier, or null if not specified.</value>
+        public string? Runtime => _runtime;
+
+        /// <summary>
+        /// Gets the target architecture (e.g., "x64", "x86", "arm64").
+        /// </summary>
+        /// <value>The architecture identifier, or an empty string if not specified.</value>
+        public string? Architecture => _architecture;
+
+        /// <summary>
+        /// Gets the path to the dotnet host executable.
+        /// </summary>
+        /// <value>The dotnet host path, or null if not specified.</value>
+        public string? DotnetHostPath => _dotnetHostPath;
+
+        /// <summary>
+        /// Gets the path to the MSBuild assembly.
+        /// </summary>
+        /// <value>The MSBuild assembly path, or null if not specified.</value>
+        public string? MSBuildAssemblyPath => _msBuildAssemblyPath;
+
+        /// <summary>
+        /// Gets if Task Host Factory was requested explicitly.
+        /// </summary>
+        public bool? IsTaskHostFactory => _isTaskHostFactory;
+
+        /// <summary>
+        /// Gets a value indicating whether returns true if parameters were unset.
+        /// </summary>
+        internal bool IsEmpty => Equals(Empty);
+
+        /// <summary>
+        /// Merges two TaskHostParameters instances, with the second parameter values taking precedence when both are specified.
+        /// </summary>
+        /// <param name="baseParameters">The base parameters.</param>
+        /// <param name="overrideParameters">The override parameters that take precedence.</param>
+        /// <returns>A new TaskHostParameters with merged values.</returns>
+        internal static TaskHostParameters MergeTaskHostParameters(TaskHostParameters baseParameters, TaskHostParameters overrideParameters)
+        {
+            // If both are empty, return empty
+            if (baseParameters.IsEmpty && overrideParameters.IsEmpty)
+            {
+                return Empty;
+            }
+
+            // If override is empty, return base
+            if (overrideParameters.IsEmpty)
+            {
+                return baseParameters;
+            }
+
+            // If base is empty, return override
+            if (baseParameters.IsEmpty)
+            {
+                return overrideParameters;
+            }
+
+            // Merge: override values take precedence, fall back to base values
+            return new TaskHostParameters(
+                runtime: overrideParameters.Runtime ?? baseParameters.Runtime,
+                architecture: overrideParameters.Architecture ?? baseParameters.Architecture,
+                dotnetHostPath: overrideParameters.DotnetHostPath ?? baseParameters.DotnetHostPath,
+                msBuildAssemblyPath: overrideParameters.MSBuildAssemblyPath ?? baseParameters.MSBuildAssemblyPath,
+                isTaskHostFactory: overrideParameters.IsTaskHostFactory ?? baseParameters.IsTaskHostFactory);
+        }
+    }
```

Comment:
My original comment was going to be "don't check `IsEmpty` twice, save it off" but you can do that and be a bit easier to understand as a pattern match:

```suggestion
            return (baseParameters.IsEmpty, overrideParameters.IsEmpty) switch
            {
                (true, true) => Empty,
                (false, true) => baseParameters,
                (true, false) => overrideParameters,
                _ => new TaskHostParameters(
                    runtime: overrideParameters.Runtime ?? baseParameters.Runtime,
                    architecture: overrideParameters.Architecture ?? baseParameters.Architecture,
                    dotnetHostPath: overrideParameters.DotnetHostPath ?? baseParameters.DotnetHostPath,
                    msBuildAssemblyPath: overrideParameters.MSBuildAssemblyPath ?? baseParameters.MSBuildAssemblyPath,
                    isTaskHostFactory: overrideParameters.IsTaskHostFactory ?? baseParameters.IsTaskHostFactory)
            };

```